app/controllers/warehouse_reports/health/eligibility_controller.rb

```diff
@@ -31,10 +31,13 @@ def create
     def update
       begin
         @report = inquiry_scope.find(params[:id].to_i)
-        @report.result = update_params[:content].read
-        @report.save!
-      rescue
-        flash[:error] = 'Error processing uploaded file'
+        Health::EligibilityResponse.create(eligibility_inquiry: @report,
+          response: update_params[:content].read,
+          user: current_user,
+          original_filename: update_params[:content].original_filename)
+        Health::FlagIneligiblePatientsJob.perform_later(@report)
```

Comment:
Generally passing an id to `perform_later` is better than an object, then reify it in the job.  The object gets serialized in the job, and depending on the length of time it takes to run the job, it might be stale.  Also, it might be large, which gums up the delayed job table.